samplers/aws/xray/rand.go

```diff
@@ -0,0 +1,114 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package xray
+
+import (
+	crand "crypto/rand"
+	"encoding/binary"
+	"math/rand"
+	"sync"
+	"time"
+)
+
+var _ rand.Source = (*lockedSource)(nil)
+var _ rand.Source64 = (*lockedSource64)(nil)
+
+type lockedSource struct {
```

Comment:
I am surprised we need an entire locked random to compute jitter. I don't see anything like it in this other library, do we really need this?

https://github.com/lthibault/jitterbug

Reply:
I have tried to add all the methods for this. Not sure using this lib is a good idea since we have like custom implementations. @Aneurysm9 any thoughts?

Reply:
Jitterbug appears to depend on a `rand.Source` implementation, which is how it ignores concurrency issues.  It pushes that off to the source.  The [default source](https://pkg.go.dev/math/rand#pkg-overview) used by the package-global functions in the `math/rand` package is safe for concurrent use, but sources created with `NewSource()` are not and need locking like this to be made safe for concurrent use.  

I suppose the question here is whether the default source can be used.

Reply:
Okay then if `jitterbug` is safe to use considering concurrency then show we remove this logic?

Reply:
Ah I was referencing jitterbug just to look at a different implementation which looked much simpler. I don't think we need to use it as a dependency but if the default source can work then we should just use that I guess

Reply:
Using the default source in a library is generally avoided because it needs to be seeded to be effectively used in most cases but, as a library author, you shouldn't be making the choice to seed it for an application that might not want it seeded and you also can't guarantee that another library doesn't come along later and re-seed it with a fixed value.  I do really wish that new sources were also safe for concurrent access.

Can the problem be avoided by simply having more sources?  Can we have each actor with its own source and thus avoid concurrency concerns?